python/pyspark/sql/functions.py

```diff
@@ -1848,6 +1848,12 @@ def __del__(self):
             self._broadcast = None
 
     def __call__(self, *cols):
+        for c in cols:
+            if not isinstance(c, (Column, str)):
+                raise TypeError(
+                    "All arguments should be Columns or strings representing column names. "
```

Comment:
Sounds good. I think it should also provide a suggestion that one can use literals (`lit`, `array`, `struct` and `create_map`).